scripts/run_mypy_checks.py

```diff
@@ -769,37 +774,65 @@ def install_third_party_libraries(skip_install):
         install_third_party_libs.main()
 
 
-def get_mypy_cmd(files):
+def get_mypy_cmd(files, ci_check):
     """Return the appropriate command to be run.
 
     Args:
         files: list(list(str)). List having first element as list of string.
+        ci_check: bool. Whether the test is for CI.
 
     Returns:
         list(str). List of command line arguments.
     """
+    # TODO(#13113): Change mypy command to mypy path after Python3 migration.
+    if ci_check:
+        mypy_cmd = 'mypy'
+    else:
+        mypy_cmd = os.path.join(
+            os.getcwd(), 'third_party', 'python3_libs', 'bin', 'mypy')
     if files:
-        cmd = [MYPY_CMD, '--config-file', CONFIG_FILE_PATH] + files
+        cmd = [mypy_cmd, '--config-file', CONFIG_FILE_PATH] + files
     else:
         excluded_files_regex = (
             '|'.join(NOT_FULLY_COVERED_FILES + EXCLUDED_DIRECTORIES))
         cmd = [
-            MYPY_CMD, '--exclude', excluded_files_regex,
+            mypy_cmd, '--exclude', excluded_files_regex,
             '--config-file', CONFIG_FILE_PATH, '.'
         ]
     return cmd
 
 
-def install_mypy_prerequisites():
+def install_mypy_prerequisites(ci_check):
     """Install mypy and type stubs from mypy_requirements.txt.
 
+    Args:
+        ci_check: bool. Whether the test is for CI.
+
     Returns:
         int. The return code from installing prerequisites.
     """
-    cmd = [PYTHON3_CMD, '-m', 'pip', 'install', '-r', MYPY_REQUIREMENTS_PATH]
-    process = subprocess.call(
-        cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
-    return process
+    # TODO(#13398): Change MyPy installation after Python3 migration.
+    if ci_check:
+        cmd = [
+            PYTHON3_CMD, '-m', 'pip', 'install', '-r',
+            MYPY_REQUIREMENTS_FILE_PATH
+        ]
+    else:
+        cmd = [
+            PYTHON3_CMD, '-m', 'pip', 'install', '-r',
+            MYPY_REQUIREMENTS_FILE_PATH, '--target', MYPY_TOOLS_DIR,
+            '--upgrade'
+        ]
+    process = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output = process.communicate()
+    if 'can\'t combine user with prefix' in output[1]:
```

Comment:
Can you add a comment explaining what case this handles?

Reply:
Actually I copied and pasted from [here](https://github.com/oppia/oppia/blob/2d1496b33ac891c40709017ffa29a9df870540e1/scripts/install_third_party_libs.py#L48).

Reply:
OK. I've reached out to @DubeySandeep to try and figure out what this is doing